File: frontend/src/app/services/roster.service.ts
import {rosterModel} from "../modules/models/rosterModel";
import {Observable} from "rxjs";
import {rosterOrErrorModel} from "../modules/models/rosterOrErrorModel";

export interface RosterService {
  saveRoster(roster: rosterModel): Observable<rosterOrErrorModel>;

  findAll(): Observable<rosterModel[]>;

  findByPlayerId(playerId: string): Observable<rosterModel>;

  deleteRecord(playerId: string): Observable<void>;

  editRecord(rosterModel: rosterModel): Observable<rosterModel>;
}

File: be/src/main/java/com/netcracker/edu/backend/service/RosterService.java
package com.netcracker.edu.backend.service;

import com.netcracker.edu.backend.entity.RosterEntity;

import java.util.List;

public interface RosterService {
    RosterEntity saveRoster(RosterEntity rosterEntity);

    boolean isExistById(String rosterId);

    List findAll();

    RosterEntity findRecord(String playerId);

    void deleteRecord(String playerId);

    void editRecord(RosterEntity rosterEntity);
}
